scripts/globals/job_utils/dancer.lua

```diff
@@ -334,64 +423,58 @@ xi.job_utils.dancer.useViolentFlourishAbility = function(player, target, ability
     end
 end
 
-local waltzAbilities =
-{
---  Spell ID                           { tpCost, statMultiplier, baseHp }
-    [xi.jobAbility.CURING_WALTZ    ] = { 200, 0.25,  60 },
-    [xi.jobAbility.CURING_WALTZ_II ] = { 350, 0.50, 130 },
-    [xi.jobAbility.CURING_WALTZ_III] = { 500, 0.75, 270 },
-    [xi.jobAbility.CURING_WALTZ_IV ] = { 650, 1.00, 450 },
-    [xi.jobAbility.CURING_WALTZ_V  ] = { 800, 1.25, 600 },
-    [xi.jobAbility.DIVINE_WALTZ    ] = { 400, 0.25,  60 },
-    [xi.jobAbility.DIVINE_WALTZ_II ] = { 800, 0.75, 270 },
-}
+xi.job_utils.dancer.useBuildingFlourishAbility = function(player, target, ability)
+    local flourishMerits = player:getMerit(xi.merit.BUILDING_FLOURISH_EFFECT)
+    local availableMoves = player:getStatusEffect(xi.effect.FINISHING_MOVE_1):getPower()
 
-xi.job_utils.dancer.checkWaltzAbility = function(player, target, ability)
-    local waltzInfo = waltzAbilities[ability:getID()]
+    local power = utils.clamp(availableMoves, 0, 3)
 
-    if target:getHP() == 0 then
-        return xi.msg.basic.CANNOT_ON_THAT_TARG, 0
-    elseif player:hasStatusEffect(xi.effect.SABER_DANCE) then
-        return xi.msg.basic.UNABLE_TO_USE_JA2, 0
-    elseif player:hasStatusEffect(xi.effect.TRANCE) then
-        ability:setRecast(math.min(ability:getRecast(), 6))
+    player:addStatusEffect(xi.effect.BUILDING_FLOURISH, power, 0, 60, 0, flourishMerits)
+    setFinishingMoves(player, availableMoves - 3)
```

Comment:
This is attempting to always remove 3 which should not be the case.  setFinishingMoves does not clamp the lower bound, so if we want to assume and remove more, that function will need additional safety.

(This one's important)

Reply:
Fixed